src/cmd/linuxkit/run_qemu.go

```diff
@@ -80,19 +80,59 @@ func runQemu(args []string) {
 	remArgs := flags.Args()
 
 	if len(remArgs) == 0 {
-		fmt.Println("Please specify the prefix to the image to boot")
+		fmt.Println("Please specify the path to the image to boot")
 		flags.Usage()
 		os.Exit(1)
 	}
-	prefix := remArgs[0]
+	path := remArgs[0]
 
-	// Print warning if conflicting UEFI and ISO flags are set
-	if *uefiBoot && *isoBoot {
-		log.Warnf("Both -iso and -uefi have been used")
+	// if user specified a path that exists and ends in .iso they meant an ISO
+	_, err := os.Stat(path)
+	stat := err == nil
+	if stat {
+		if !*isoBoot && strings.HasSuffix(path, ".iso") {
+			*isoBoot = true
+		}
+		if strings.HasSuffix(path, "-efi.iso") {
+			*uefiBoot = true
+		}
+	} else {
+		// old behaviour of allowing a prefix for non kernel outputs
+		if *isoBoot && !*uefiBoot {
+			_, err := os.Stat(path + ".iso")
+			if err == nil {
+				path = path + ".iso"
+			}
+		}
+		if *uefiBoot {
+			_, err := os.Stat(path + "-efi.iso")
+			if err == nil {
+				*isoBoot = true
+				path = path + "-efi.iso"
+			}
+		}
+	}
+
+	// user not trying to boot off ISO or kernel, so assume booting from a disk image
+	if stat && !*isoBoot {
+		_, err = os.Stat(path + "-kernel")
+		statKernel := err == nil
+		if !statKernel {
+			if *disk != "" {
+				log.Fatalf("Cannot boot from disk and specify a disk as well at present")
+			}
+			*disk = path
+			*kernelBoot = false
+		}
+	}
+
+	// override default kernel boot if ISO specified
+	if *isoBoot {
+		*kernelBoot = false
```

Comment:
I'm not sure I like this complicated scheme. It's really hard to describe in words, so probably not right...if we just ditch the old behaviour when it comes to ISO or UEFI boot it'll be simpler.